test/unit/test_table.c

```diff
@@ -106,15 +106,157 @@ bool test_r_table_sort1(void) {
 	mu_end;
 }
 
-bool test_r_table_columns() {
+bool test_r_table_uniq(void) {
+	RTable *t = __table_test_data1 ();
+
+	r_table_uniq (t);
+	char *strd = r_table_tostring (t);
+	mu_assert_streq (strd,
+		"ascii code\n"
+		"----------\n"
+		"a     97\n"
+		"b     98\n"
+		"c     99\n",
+		"uniq delete nothing");
+	free (strd);
+
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "d", "99", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "d", "99", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "c", "100", NULL);
+
+	r_table_uniq (t);
+	char *stri = r_table_tostring (t);
+	mu_assert_streq (stri,
+		"ascii code\n"
+		"----------\n"
+		"a     97\n"
+		"b     98\n"
+		"c     99\n"
+		"d     99\n"
+		"c     100\n",
+		"uniq delete some rows");
+	free (stri);
+	r_table_free (t);
+	mu_end;
+}
+
+static void merge_second(RList *acc, RList *new_row, int nth) {
```

Comment:
maybe even better to directly pass `RTableRow` instead of just the `items` part.

Reply:
also, i'm not sure `nth` is really necessary as I can imagine when you implement the merge function you already know the column(s) you want to work on.

Reply:
```nth``` could help to create reusable function.
For example a common merge function with a default separator.